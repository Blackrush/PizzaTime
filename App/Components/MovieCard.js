import React, { Component } from 'react'
import {
  StyleSheet,
  TouchableWithoutFeedback,
  Image,
  View,
} from 'react-native'

import StarRating from './StarRating'


class MovieCard extends Component {
  static propTypes = {
    onPress: React.PropTypes.func.isRequired,
    movie: React.PropTypes.object.isRequired,
  }

  static defaultProps = {
    onPress() {},
  }

  render() {
    const {movie: {rating, images}} = this.props

    return (
      <TouchableWithoutFeedback style={styles.card} onPress={() => this.props.onPress(this.props.movie)}>
        <Image source={{uri: images.poster.thumb}} style={styles.card}>
          <View style={styles.cardContent}>
            <StarRating rating={rating/10*5} maxStars={5}/>
          </View>
        </Image>
      </TouchableWithoutFeedback>
    )
  }
}

const styles = StyleSheet.create({
  card: {
    width: 100,
    height: 150,
    justifyContent: 'flex-end',
    alignItems: 'center',
    margin: 10,
    padding: 5,
    borderRadius: 5,
  },
  cardContent: {
    backgroundColor: 'rgba(0,0,0,0)',
    alignSelf: 'stretch',
    alignItems: 'center',
  },
})

export default MovieCard