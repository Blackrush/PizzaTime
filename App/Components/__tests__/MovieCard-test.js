import React from 'react'
import 'react-native'

import MovieCard from '../MovieCard'
import { create as render } from 'react-test-renderer'

test('Movies renders correctly', () => {
  const movie = require('./movie.json')
  const tree = render(
    <MovieCard onPress={() => false} movie={movie}/>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
