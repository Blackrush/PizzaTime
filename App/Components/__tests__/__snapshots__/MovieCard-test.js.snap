exports[`test Movies renders correctly 1`] = `
<Image
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  source={
    Object {
      "uri": "https://walter.trakt.us/images/movies/000/187/173/posters/thumb/2b19d6ec1f.jpg"
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "height": 150,
      "justifyContent": "flex-end",
      "margin": 10,
      "padding": 5,
      "width": 100
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "rgba(0,0,0,0)"
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between"
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 11,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 11,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 11,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 11,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gray",
                "fontFamily": "FontAwesome",
                "fontSize": 11,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
    </View>
  </View>
</Image>
`;
